### PR TITLE
Overhaul usage of test-runtime, Update buildtools

### PR DESCRIPTION
**Pull request by mellinoe**
This change uses some features available in an updated version of buildtools. Instead of resolving the test-runtime for each of our test projects as a separate step, we will now add a "reference" to it from the
project.json of each test project. This will allow us to resolve the nuget assets from both it and the test project at the same time. This will be more robust than our current process, which is very fragile and error-prone.

The above is made possible by an updated version of "PrereleaseResolveNuGetPackageAssets", which is the buildtools task responsible for resolving test assets from nuget packages. This task also has a lot of additional validation, which triggers a lot of warnings in corefx. I've made various changes to csproj's and to project.json's around the repository in order to satisfy those warnings.

This change will get us closer to clean outerloop runs on OSX and Ubuntu. Currently these runs are plagued by various issues. The primary issue is that many test projects are restoring test assets incorrectly, due to the separation of the test-runtime and the test's direct dependencies. My goal is that when we finally manage to get a clean outerloop run (which may require additional changes after this one), we convert one or all of the OSX/Ubuntu inner-loop CI runs to use build.sh, so that we can prevent regressions in the future.

NOTE: This should also remove our dependency on the --infer-runtimes feature of CLI (which is not going to be supported, long-term). I haven't removed that parameter as part of this change, but it should no longer be necessary, because we are specifying all of the runtimes we need in all of our test project.json's.

@ericstj , @weshaggard 

Requires https://github.com/dotnet/buildtools/pull/616 to be published, and another change here to use that version.

### CODE REVIEW

**Comment by mellinoe** [comment]
@ericstj I mentioned I needed to fix up a few project.json's because they were not referencing the netstandard version that the csproj indicated. Could you take a look at these and see if they are correct now?

* [System.Data.SqlClient](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Data.SqlClient/ref/project.json)
* [System.Diagnostics.DiagnosticSource](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj)
* [System.IO.Packaging](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.IO.Packaging/ref/project.json)
* [System.Linq.Expressions](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Linq.Expressions/ref/project.json)
* [System.Net.Http.WinHttpHandler](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Net.Http.WinHttpHandler/src/System.Net.Http.WinHttpHandler.csproj)
* [System.Net.NetworkInformation](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Net.NetworkInformation/ref/project.json)
* [System.Numerics.Vectors](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Numerics.Vectors/ref/project.json)
* [System.Reflection.TypeExtensions](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Reflection.TypeExtensions/ref/4.0.0/project.json)
* [System.Runtime.InteropServices.PInvoke](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Runtime.InteropServices.PInvoke/ref/project.json)
* [System.Runtime.InteropServices.RuntimeInformation](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Runtime.InteropServices.RuntimeInformation/src/System.Runtime.InteropServices.RuntimeInformation.csproj)
* [System.Runtime.WindowsRuntime.UI.Xaml](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Runtime.WindowsRuntime.UI.Xaml/src/project.json)
* [System.Xml.Xsl.Primitives](https://github.com/mellinoe/corefx/blob/test-runtime-projectref-rebase2/src/System.Xml.Xsl.Primitives/ref/System.Xml.Xsl.Primitives.csproj)

File: src/Common/tests/System/Diagnostics/RemoteExecutorConsoleApp/project.json [hunk 1]
```diff
@@ -1,9 +1,6 @@
 {
   "dependencies": {
     "System.Runtime": "4.1.0-rc3-24011-00",
-    "Microsoft.DotNet.xunit.performance": "1.0.0-alpha-build0029",
```
Comment (mellinoe) [inline src/Common/tests/System/Diagnostics/RemoteExecutorConsoleApp/project.json]: These weren't actually used in the test project, and were causing some build warnings due to "different version of the same dependent assembly", so I removed them from this project.

**Comment by mellinoe** [comment]
@ericstj All of the issues should be addressed now.

**Comment by mellinoe** [comment]
Buildtools is currently experiencing some issue with signing in the official build, so we can't get a new version of buildtools published yet. I'll update the PR when a new version of the package is available.

**Comment by mellinoe** [comment]
@weshaggard , @joperezr Please take a look, we will need the changes here to take a newer version of buildtools.

**Comment by weshaggard** [comment]
LGTM

**Comment by ericstj** [comment]
This LGTM.  Thanks for doing this.

**Pull request merged**